add $note_id to workflow handleNewNote method
this could be used in workflow to process note contents, for example act on
sender (check for Notification::isBounceMessage for example)

--- eventum-1.7.1/include/class.workflow.php	2006-04-06 22:44:17.388665653 +0300
+++ /home/glen/class.workflow.php	2006-04-06 22:41:44.000000000 +0300
@@ -302,14 +302,15 @@
      * @param   integer $issue_id The ID of the issue.
      * @param   integer $usr_id The user ID of the person posting this new note
      * @param   boolean $closing If the issue is being closed
+     * @param   integer $note_id The ID of the new note
      */
-    function handleNewNote($prj_id, $issue_id, $usr_id, $closing = false)
+    function handleNewNote($prj_id, $issue_id, $usr_id, $closing = false, $note_id = false)
     {
         if (!Workflow::hasWorkflowIntegration($prj_id)) {
             return;
         }
         $backend =& Workflow::_getBackend($prj_id);
-        return $backend->handleNewNote($prj_id, $issue_id, $usr_id, $closing);
+        return $backend->handleNewNote($prj_id, $issue_id, $usr_id, $closing, $note_id);
     }
 
 
--- eventum-1.7.1/include/class.note.php	2006-04-06 22:44:17.768674142 +0300
+++ /home/glen/class.note.php	2006-04-06 22:41:44.000000000 +0300
@@ -359,7 +359,7 @@
                 } else {
                     Notification::notify($issue_id, 'notes', $new_note_id, $internal_only);
                 }
-                Workflow::handleNewNote(Issue::getProjectID($issue_id), $issue_id, $usr_id, $closing);
+                Workflow::handleNewNote(Issue::getProjectID($issue_id), $issue_id, $usr_id, $closing, $new_note_id);
             }
             // need to return the new note id here so it can
             // be re-used to associate internal-only attachments
--- eventum-1.7.1/include/workflow/class.example.php	2006-04-06 22:44:17.388665653 +0300
+++ /home/glen/class.example.php	2006-04-06 22:41:44.000000000 +0300
@@ -113,8 +113,9 @@
      * @param   integer $issue_id The ID of the issue.
      * @param   integer $usr_id The user ID of the person posting this new note
      * @param   boolean $closing If the issue is being closed
+     * @param   integer $note_id The ID of the new note
      */
-    function handleNewNote($prj_id, $issue_id, $usr_id, $closing)
+    function handleNewNote($prj_id, $issue_id, $usr_id, $closing, $note_id)
     {
         echo "Workflow: New Note<br />\n";
     }
--- eventum-1.7.1/include/workflow/class.abstract_workflow_backend.php	2006-04-06 22:44:17.388665653 +0300
+++ /home/glen/class.abstract_workflow_backend.php	2006-04-06 22:41:44.000000000 +0300
@@ -106,8 +106,9 @@
      * @param   integer $issue_id The ID of the issue.
      * @param   integer $usr_id The user ID of the person posting this new note
      * @param   boolean $closing If the issue is being closed
+     * @param   integer $note_id The ID of the new note
      */
-    function handleNewNote($prj_id, $issue_id, $usr_id, $closing)
+    function handleNewNote($prj_id, $issue_id, $usr_id, $closing, $note_id)
     {
     }
 
